packages/@aws-cdk/aws-ec2/lib/volume.ts

```diff
@@ -210,4 +213,344 @@ export enum EbsDeviceVolumeType {
+   *
+   * @default false
+   */
+  readonly encrypted?: boolean;
+
+  /**
+   * The customer-managed encryption key that is used to encrypt the Volume. The encrypted property must
+   * be true if this is provided.
+   *
+   * @default The default KMS key for the account, region, and EC2 service is used.
+   */
+  readonly encryptionKey?: IKey;
+
+  /**
+   * Indicates whether the volume is auto-enabled for I/O operations. By default, Amazon EBS disables I/O to the volume from attached EC2
+   * instances when it determines that a volume's data is potentially inconsistent. If the consistency of the volume is not a concern, and
+   * you prefer that the volume be made available immediately if it's impaired, you can configure the volume to automatically enable I/O.
+   *
+   * @default false
+   */
+  readonly autoEnableIo?: boolean;
+
+  /**
+   * The type of the volume; what type of storage to use to form the EBS Volume.
+   *
+   * @default {@link EbsDeviceVolumeType.GENERAL_PURPOSE_SSD}
+   */
+  readonly volumeType?: EbsDeviceVolumeType;
+
+  /**
+   * The number of I/O operations per second (IOPS) to provision for the volume, with a maximum ratio of 50 IOPS/GiB.
+   * See {@link https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ebs-volume-types.html#EBSVolumeTypes_piops|Provisioned IOPS SSD (io1) volumes}
+   * for more information.
+   *
+   * This parameter is valid only for PROVISIONED_IOPS_SSD volumes.
+   *
+   * @default None -- Required for {@link EbsDeviceVolumeType.PROVISIONED_IOPS_SSD}
+   */
+  readonly iops?: number;
+}
+
+/**
+ * Attributes required to import an existing EBS Volume into the Stack.
+ */
+export interface VolumeAttributes {
+  /**
+   * The EBS Volume's ID
+   */
+  readonly volumeId: string;
+
+  /**
+   * The availability zone that the EBS Volume is contained within (ex: us-west-2a)
+   */
+  readonly availabilityZone: string;
+
+  /**
+   * The customer-managed encryption key that is used to encrypt the Volume.
+   *
+   * @default None -- The EBS Volume is not using a customer-managed KMS key for encryption.
+   */
+  readonly encryptionKey?: IKey;
+}
+
+/**
+ * Common behavior of Volumes. Users should not use this class directly, and instead use ``Volume``.
+ */
+abstract class VolumeBase extends Resource implements IVolume {
+  public abstract readonly volumeId: string;
+  public abstract readonly availabilityZone: string;
+  public abstract readonly encryptionKey?: IKey;
+
+  public grantAttachVolume(grantee: IGrantable, instance?: IInstance): Grant {
+    const stack = Stack.of(this);
+    const result = Grant.addToPrincipal({
+      grantee,
+      actions: [ 'ec2:AttachVolume' ],
+      resourceArns: [
+        `arn:aws:ec2:${stack.region}:${stack.account}:instance/${instance?.instanceId ?? '*'}`,
+        `arn:aws:ec2:${stack.region}:${stack.account}:volume/${this.volumeId}`,
+      ],
+    });
+
+    /*
+    if (instance) {
+      this.encryptionKey?.grantEncryptDecrypt(instance);
```

Comment:
?

Reply:
Commented out because it turns out that we don't need it. I've removed the code.